src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -524,8 +524,9 @@ impl<'a> Linker for GccLinker<'a> {
             return;
         }
 
+        let is_windows = self.sess.target.target.options.is_like_windows;
         let mut arg = OsString::new();
-        let path = tmpdir.join("list");
+        let path = tmpdir.join(if is_windows { "list.def" } else { "list" });
```

Comment:
Nit: it can be `list.def` in both cases?

Reply:
I'm not sure if this won't break linkers on Unix but I'll check tomorrow.

Reply:
Seems to work just fine on Linux but looking at the docs version-scripts typically use `.map` extension.
What should I do here?

Reply:
On Windows `.map` files have a different meaning. https://docs.microsoft.com/en-us/cpp/build/reference/map-generate-mapfile?view=vs-2019

Reply:
On Windows MinGW linkers won't work unless that file has `.def` extension so it's out of the question here.

Reply:
Ok, if there are any doubts then it can stay as is.